mmdet/structures/bbox/base_bbox.py

```diff
@@ -0,0 +1,523 @@
+            dim (int): Dimension to insert. Defaults to 0.
+
+        Returns:
+            T: Concatenated box instance.
+        """
+        assert isinstance(bbox_list, Sequence)
+        if len(bbox_list) == 0:
+            raise ValueError('bbox_list should not be a empty list.')
+
+        assert dim != -1 and dim != bbox_list[0].dim()
+        assert all(isinstance(bboxes, cls) for bboxes in bbox_list)
+
+        th_bbox_list = [bboxes.tensor for bboxes in bbox_list]
+        return cls(torch.stack(th_bbox_list, dim=dim), clone=False)
+
+    @abstractproperty
+    def centers(self) -> Tensor:
+        """Return a tensor representing the centers of boxes."""
+        pass
+
+    @abstractproperty
+    def areas(self) -> Tensor:
+        """Return a tensor representing the areas of boxes."""
+        pass
+
+    @abstractproperty
+    def widths(self) -> Tensor:
+        """Return a tensor representing the widths of boxes."""
+        pass
+
+    @abstractproperty
+    def heights(self) -> Tensor:
+        """Return a tensor representing the heights of boxes."""
+        pass
+
+    @abstractmethod
+    def flip_(self,
+              img_shape: Tuple[int, int],
+              direction: str = 'horizontal') -> None:
+        """Inplace flip bboxes horizontally or vertically.
+
+        Args:
+            img_shape (Tuple[int, int]): A tuple of image height and width.
+            direction (str): Flip direction, options are "horizontal",
+                "vertical" and "diagonal". Defaults to "horizontal"
+        """
+        pass
+
+    @abstractmethod
+    def translate_(self, distances: Tuple[float, float]) -> None:
+        """Inplace translate bboxes.
+
+        Args:
+            distances (Tuple[float, float]): translate distances. The first
+                is horizontal distance and the second is vertical distance.
+        """
+        pass
+
+    @abstractmethod
+    def clip_(self, img_shape: Tuple[int, int]) -> None:
+        """Inplace clip boxes according to the image shape.
+
+        Args:
+            img_shape (Tuple[int, int]): A tuple of image height and width.
+        """
+        pass
+
+    @abstractmethod
+    def rotate_(self, center: Tuple[float, float], angle: float) -> None:
+        """Inplace rotate all boxes.
+
+        Args:
+            center (Tuple[float, float]): Rotation origin.
+            angle (float): Rotation angle represented in degrees. Positive
+                values mean clockwise rotation.
+        """
+        pass
+
+    @abstractmethod
+    def project_(self, homography_matrix: Union[Tensor, np.ndarray]) -> None:
+        """Inplace geometric transformation for bbox.
+
+        Args:
+            homography_matrix (Tensor or np.ndarray]):
+                Shape (3, 3) for geometric transformation.
+        """
+        pass
+
+    @abstractmethod
+    def rescale_(self, scale_factor: Tuple[float, float]) -> None:
+        """Inplace rescale boxes w.r.t. rescale_factor.
```

Comment:
Rescale boxes w.r.t. ``rescale_factor`` in-place.
Also change the style of description in resize_ and other in-place functions to 
`` Do something xxx in-place.``